fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -105,12 +106,24 @@ def fit(self, X, y, **kwargs):
         :param sensitive_features: A (currently) required keyword argument listing the
             feature used by the constraints object
         :type sensitive_features: numpy.ndarray, pandas.DataFrame, pandas.Series, or list (for now)
+
+        :param verbose: An optional argument to enable/disable the progress bar.
+            Default: False
+        :type verbose: bool
         """
         self.predictors_ = []
         self.lambda_vecs_ = pd.DataFrame(dtype=np.float64)
         self.objectives_ = []
         self.gammas_ = pd.DataFrame(dtype=np.float64)
         self.oracle_execution_times_ = []
+        verbose = kwargs.pop('verbose', False)
+
+        if verbose:
+            # We need tqdm to show the progress bar.
+            try:
+                from tqdm import tqdm
+            except ImportError:
+                raise RuntimeError(_TQDM_IMPORT_ERROR_MESSAGE)
```

Comment:
Rather than raise an exception, since `verbose=True` is the default, I think it would be kinder to log a warning and behave as if `verbose` was false.

Tagging @adrinjalali , who first raised a caution around `tqdm`

Reply:
I set verbose default to False so that the user ia not forced to install tqdm.